src/dvc_data/index/index.py

```diff
@@ -133,6 +133,8 @@ class Storage:
     # fs/path pair if we have the file stashed somewhere
     fs: Optional["FileSystem"] = None
     path: Optional[str] = None
+    remote_fs: Optional["FileSystem"] = None
+    remote_path: Optional[str] = None
```

Comment:
Though might want to bring something like dvc's `Remote` class https://github.com/iterative/dvc/blob/6dccfcb3f9162ef7a0b197818c87265cff9ad1f9/dvc/data_cloud.py#L20 here later on. E.g. to do `storage.remote.odb` and `storage.remote.fs/path`